src/js/libcs/OpImageDrawing.js

```diff
@@ -388,3 +388,167 @@ evaluator.cameravideo$0 = function() {
     });
     return img;
 };
+
+var helpercanvas; //invisible helper canvas.
+/**
+ * reads a rectangular block of pixels from the upper left corner.
+ * The colors are representent as a 4 component RGBA vector with entries in [0,1]
+ */
+function readPixelsIndirection(img, x, y, width, height) {
+    var res = [];
+    if (img.readPixels) {
+        res = img.readPixels(x, y, width, height);
+    } else { //use canvas-approach
+        var data, ctx;
+        if (img.img.getContext) { //img is a canvas
+            ctx = img.img.getContext('2d');
+            data = ctx.getImageData(x, y, width, height).data;
+        } else { //copy corresponding subimage of img.img to temporary canvas
+            if (!helpercanvas) {
+                //creating helpercanvas only once increases the running time
+                helpercanvas = /** @type {HTMLCanvasElement} */ (document.createElement("canvas"));
+            }
+            helpercanvas.width = width;
+            helpercanvas.height = height;
+
+            ctx = helpercanvas.getContext('2d');
+            ctx.drawImage(img.img, x, y, width, height, 0, 0, width, height);
+            data = ctx.getImageData(0, 0, width, height).data;
+        }
+        for (var i in data) res.push(data[i] / 255);
+    }
+    return res;
+}
+
+/**
+ * imagergba(‹image›,x,y) implements imagergb(‹imagename›,x,y) from Cinderella, i.e.
+ * returns a 4 component vector ranging from (0-255, 0-255, 0-255, 0-1)
+ */
+evaluator.imagergba$3 = function(args, modifs) {
+    var img = imageFromValue(evaluateAndVal(args[0]));
+    var x = evaluateAndVal(args[1]);
+    var y = evaluateAndVal(args[2]);
+
+    if (!img || x.ctype !== 'number' || y.ctype !== 'number') return nada;
```

Comment:
:bomb: I think `!img.ready` should be another reason to return `nada`. Both here and in the other result from `imageFromValue`.